packages/shell-api/src/replica-set.ts

```diff
@@ -18,15 +19,49 @@ export default class ReplicaSet extends ShellApiClass {
   constructor(mongo) {
     super();
     this._mongo = mongo;
-    const proxy = new Proxy(this, {
-      get: (obj, prop): any => {
-        if (!(prop in obj)) {
-          throw new MongoshUnimplementedError('rs not currently supported');
-        }
-        return obj[prop];
-      }
-    });
-    return proxy;
+  }
+
+  /**
+   *  rs.initiate calls replSetInitiate admin command.
+   *
+   * @param config
+   */
+  @returnsPromise
+  async initiate(config?: any): Promise<any> {
+    this._emitReplicaSetApiCall('initiate', { config });
+    return this._mongo._serviceProvider.runCommandWithCheck(ADMIN_DB, { replSetInitiate: config });
+  }
+
+  /**
+   *  rs.config calls replSetReconfig admin command.
+   *
+   *  Returns a document that contains the current replica set configuration.
+   */
+  @returnsPromise
+  async config(): Promise<any> {
+    this._emitReplicaSetApiCall('config', {});
+    return this._mongo._serviceProvider.runCommandWithCheck(ADMIN_DB, { replSetReconfig: 1 });
+  }
+
+  /**
+   *  rs.reconfig calls replSetReconfig admin command.
+   *
+   *  @param config
+   *  @param options
+   */
+  @returnsPromise
+  async reconfig(config: any, options?: any): Promise<any> {
```

Comment:
```suggestion
  async reconfig(config: object, options?: object): Promise<any> {
```

Reply:
I totally agree with replacing `any`, although `object` is not what we want (it only matches `{}`). `Record<string, any>` could be an option.

Reply:
> Agree with replacing `any`, although `object` is not what we want (it only matches `{}`)

Right, but `{}` matches a superset of all objects. :) I think the difference between it any `Record<string, any>` is basically that the latter excludes symbol keys, which also seems fine to me.

Reply:
> {} matches a superset of all objects

Oh no .. i mean `object` in typescript only matches an empty object. That it would match any object is what anyone would expect .. for some reason it doesn't seem to be the case.

```
function foo1(x: object) {
  return x.bar; // complains
}

function foo2(x: Object) {
  return x.bar; // complains
}

function foo3(x: {[key: string]: any}) {
  return x.bar; // ok
}

function foo4(x: Record<string, any>) {
  return x.bar; // ok
}
```

https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABMOcCMAKAHgLkXAIwCsBTaASkQG8AoRRAJxKhAaSwDoCBDBgbhoBfGjVCRYCZKgBM2PAHliZKJVr0mLNok49+QkWOjwkKOAGY51ANoBrEgE88AZygMYYAOYBdPNzD3BVTpGZlZ2Ll4BYVFwI0lTABZLACUyOAYAEwAeFzdPABpEP3sAPiD1UK0dSKEgA